Introduce withRefreshAuth (#8574)
marmelab/react-admin: docs/addRefreshAuthToDataProvider.md

---
layout: default
title: "addRefreshAuthToDataProvider"
---

# `addRefreshAuthToDataProvider`

This helper function wraps an existing [`dataProvider`](./DataProviderIntroduction.md) to support authentication token refreshing mechanisms.

## Usage

Use `addRefreshAuthToDataProvider` to decorate an existing data provider. In addition to the base provider, this function takes a function responsible for refreshing the authentication token if needed.

> Here is a simple example that refreshes an expired JWT token when needed:

Reviewer: I wonder if we should add this simple example as its own dedicated paragraph in the Security > Introduction page too. What do you think @fzaninotto ?

Reviewer 2: 👍

Author: Done

}

// in src/dataProvider.js
import { addRefreshAuthToDataProvider } from 'react-admin';
import simpleRestProvider from 'ra-data-simple-rest';
import { refreshAuth } from 'refreshAuth';

const baseDataProvider = simpleRestProvider('http://path.to.my.api/');

export const dataProvider = addRefreshAuthToDataProvider(baseDataProvider, refreshAuth);
```

> Then, inject the decorated provider in the `<Admin>` component:

Author: ```suggestion
Then, pass the decorated provider to the `<Admin>` component
```
